source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -465,6 +471,24 @@ TemporaryFileStream CopyStreamToFile(Guid id, long length, BinaryReader reader)
             }
             return new TemporaryFileStream(path, log);
         }
+        
+        async Task<TemporaryFileStream> CopyStreamToFileAsync(Guid id, long length, Stream stream, CancellationToken cancellationToken)
+        {
+            var path = Path.Combine(Path.GetTempPath(), string.Format("{0}_{1}", id.ToString(), Interlocked.Increment(ref streamCount)));
+            long bytesLeftToRead = length;
+            using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
+            {
+                var buffer = new byte[65*1024];
```

Comment:
The buffer size here is different to the sync version. Is that done to keep the size < LOH threshold?

Reply:
yes